tests/test_approvals.py

```diff
@@ -9,6 +9,129 @@
 # testing this here is more convenient
 from marge.job import CannotMerge, _get_reviewer_names_and_emails
 
+CODEOWNERS = {
+    "content": """
+# This is an example code owners file, lines starting with a `#` will
+# be ignored.
+
+* @default-codeowner @test-user1
+* @test-user1 @ebert
+
+unmatched/* @test5
+"""
+}
+
+# pylint: disable=anomalous-backslash-in-string
+CODEOWNERS_FULL = {
+    "content": """
+# This is an example code owners file, lines starting with a `#` will
+# be ignored.
+
+# app/ @commented-rule
+
+# We can specify a default match using wildcards:
+* @default-codeowner
+
+# Rules defined later in the file take precedence over the rules
+# defined before.
+# This will match all files for which the file name ends in `.rb`
+*.rb @ruby-owner
+
+# Files with a `#` can still be accesssed by escaping the pound sign
+\#file_with_pound.rb @owner-file-with-pound
+
+# Multiple codeowners can be specified, separated by spaces or tabs
+CODEOWNERS @multiple @code @owners
+
+# Both usernames or email addresses can be used to match
```

Comment:
Reading the code only usernames not email addresses work.

Reply:
This might fall into the same category as the group comment, I wonder what would be the best way to support emails or whether it's necessary? AFAIK gitlab only supports `@username` format in the comments so it would take another lookup to the users api to find username for the comment that gets added to the MR. It might be simpler than it sounds, will have a look at it